spec/struct.dd

```diff
@@ -1373,6 +1373,237 @@ $(H3 $(LNAME2 implicit-copy-constructors, Implicit Copy Constructors))
+
+    $(COMMENT $(P A `struct` that defines a move constructor
+    is not $(RELATIVE_LINK2 POD, POD).))
+    $(P If a move constructor is declared, also declare a copy constructor.)
+
+    $(P A constructor declaration is a move constructor declaration if it meets
+    the following requirements:)
+
+    $(UL
+    $(LI It takes exactly one parameter without a
+    $(DDSUBLINK spec/function, function-default-args, default argument),
+    followed by any number of parameters with default arguments.)
+
+    $(LI Its first parameter is not a
+    $(DDSUBLINK spec/function, ref-params, `ref` parameter).)
+
+    $(LI The type of its first parameter is the same type as
+    $(DDSUBLINK spec/type, typeof, `typeof(this)`), optionally with one or more
+    $(DDLINK spec/const3, Type Qualifiers, type qualifiers) applied to it.)
+
+    $(LI It is not a
+    $(DDSUBLINK spec/template, template_ctors, template constructor declaration).)
+    )
+
+    ---
+    struct A
+    {
+        this(ref return scope A rhs) {}                    // copy constructor
+        this(return scope A rhs) {}                        // move constructor
+        this(return scope const A rhs, int b = 7) {}       // move constructor with default parameter
+    }
+    ---
+
+    $(P The move constructor is type checked as a normal constructor.)
+
+    $(P The move constructor's first parameter only accepts rvalues. An
+    lvalue can be coerced into being an rvalue using
+    $(DDSUBLINK spec/expression, RvalueExpression, `__rvalue(Expression)`).)
+
+    $(P If a move constructor is defined, implicit calls to it will be inserted
+    in the following situations:)
+
+    $(OL
+    $(LI When a variable is explicitly initialized:)
+
+    $(SPEC_RUNNABLE_EXAMPLE_RUN
+    ---
+    struct A
+    {
+        int[] arr;
+        this(return scope A rhs) // move constructor
+        {
+            arr = rhs.arr;
+            rhs.arr = null; // do not leave dangling reference to array
+        }
+        this(ref return scope A rhs) { assert(0); }
+    }
+
+    void main()
+    {
+        A a;
+        a.arr = [1, 2];
+
+        A b = __rvalue(a); // move constructor gets called
+        b.arr[] += 1;
+        assert(a.arr is null); // a.arr is gone
+        assert(b.arr == [2, 3]);
+    }
+    ---
+    )
+
+    $(LI When a parameter is passed by value to a function:)
+
+    $(SPEC_RUNNABLE_EXAMPLE_COMPILE
+    ---
+    struct A
+    {
+        this(return scope A another) {} // move constructor
+        this(ref return scope A rhs) { assert(0); }
+    }
+
+    void fun(A a) {}
+
+    void main()
+    {
+        A a;
+        fun(__rvalue(a)); // no constructor gets called
+        // `a` is no longer a valid object
+    }
+    ---
+    )
+)
+
+$(H3 $(LNAME2 disable-move, Disabled Moving))
+
+    $(P When a move constructor is defined for a `struct` (or marked `@disable`), the compiler no
+    longer implicitly generates default move constructors for that `struct`:
```

Comment:
I suggest a link to the relevant section which the reader won't have seen yet.
```suggestion
    longer $(RELATIVE_LINK2 implicit-move-constructors, implicitly generates) default move constructors for that `struct`:
```

Reply:
added